geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ClientProtocolServiceLoader.java

```diff
@@ -18,17 +18,23 @@
 import java.util.Iterator;
 import java.util.ServiceLoader;
 
-public class MessageHandlerFactory {
-  public ClientProtocolMessageHandler makeMessageHandler() {
-    ServiceLoader<ClientProtocolMessageHandler> loader =
-        ServiceLoader.load(ClientProtocolMessageHandler.class);
-    Iterator<ClientProtocolMessageHandler> iterator = loader.iterator();
+public class ClientProtocolServiceLoader {
+  public ClientProtocolService loadService() {
+    ServiceLoader<ClientProtocolService> loader = ServiceLoader.load(ClientProtocolService.class);
+    Iterator<ClientProtocolService> iterator = loader.iterator();
 
     if (!iterator.hasNext()) {
       throw new ServiceLoadingFailureException(
-          "There is no ClientProtocolMessageHandler implementation found in JVM");
+          "There is no ClientProtocolService implementation found in JVM");
     }
 
-    return iterator.next();
+    ClientProtocolService service = iterator.next();
+
+    if (iterator.hasNext()) {
```

Comment:
Would it be possible to get/load concern service?

Reply:
We could check a field on the service; currently we're mostly using the service to resolve circular dependencies and don't expect users to implement it.